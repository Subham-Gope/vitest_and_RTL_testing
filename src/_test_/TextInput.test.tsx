// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import TextInput from "../components/TextInput";

// describe("Testing the TextInput.tsx file", () => {
//   test("let's enter a text", async () => {
//     const user = userEvent.setup();
//     const { container } = render(<TextInput />);
//     expect(container).toMatchSnapshot();

//     const input = screen.getByRole("textbox", { name: /enter your name/i });
//     expect(input).toBeInTheDocument();

//     // simulating user typing
//     await user.type(input, "Sigma Lord");
//     expect(input).toHaveValue("Sigma Lord");
//   });
// });
