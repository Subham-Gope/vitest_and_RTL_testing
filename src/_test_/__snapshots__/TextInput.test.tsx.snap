// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the TextInput.tsx file > let's enter a text 1`] = `
<div>
  <div>
    <label
      for="textInput"
    >
      Enter Your Name
    </label>
    <input
      aria-label="Enter Your Name"
      id="textInput"
      placeholder="Subham Gope"
      type="text"
      value=""
    />
  </div>
</div>
`;
